src/main/io/serial.c

```diff
@@ -341,6 +341,27 @@ bool isSerialConfigValid(const serialConfig_t *serialConfigToCheck)
     for (int index = 0; index < SERIAL_PORT_COUNT; index++) {
         const serialPortConfig_t *portConfig = &serialConfigToCheck->portConfigs[index];
 
+#ifdef USE_SOFTSERIAL
+        if ((portConfig->identifier == SERIAL_PORT_SOFTSERIAL1) ||
+            (portConfig->identifier == SERIAL_PORT_SOFTSERIAL2)) {
+            // Ensure MSP or serial RX is not enabled on soft serial ports
+            serialConfigToCheck->portConfigs[index].functionMask &= ~(FUNCTION_MSP | FUNCTION_RX_SERIAL);
+
+            // Ensure that the baud rate on soft serial ports is limited to 19200
+            if (portConfig->gps_baudrateIndex > BAUD_19200) {
+                serialConfigToCheck->portConfigs[index].gps_baudrateIndex = BAUD_19200;
+            }
+
+            if (portConfig->blackbox_baudrateIndex > BAUD_19200) {
+                serialConfigToCheck->portConfigs[index].blackbox_baudrateIndex = BAUD_19200;
+            }
+
+            if (portConfig->telemetry_baudrateIndex > BAUD_19200) {
+                serialConfigToCheck->portConfigs[index].telemetry_baudrateIndex = BAUD_19200;
+            }
```

Comment:
```suggestion
            portConfig->gps_baudrateIndex = constrain(portConfig->gps_baudrateIndex, BAUD_9600, BAUD_19200);
            portConfig->blackbox_baudrateIndex = constrain(portConfig->blackbox_baudrateIndex, BAUD_9600, BAUD_19200);
            portConfig->telemetry_baudrateIndex = constrain(portConfig->telemetry_baudrateIndex, BAUD_9600, BAUD_19200);
        }
```

Reply:
That would prevent selection of auto I think.

Reply:
Nice idea though, so adjusted accordingly.